### PR TITLE
devicestate: use EncryptionType

### PR DESCRIPTION
**Pull request by mvo5**
This branch uses a new `EncryptionType` in `DeviceManager.checkEncryption()` and `DeviceManager.checkFDEFeatures()`. 

This prepares for the "device-setup" feature that will not use cryptsetup but instead uses a device specific hook to configure crypto hardware.

### CODE REVIEW

**Review by mardy — APPROVED** [review]
Apart from the build failure, LGTM!

File: secboot/encrypt.go [hunk 1]
```diff
@@ -116,3 +116,12 @@ func NewAuxKey() (AuxKey, error) {
 	// On return, n == len(b) if and only if err == nil
 	return key, err
 }
+
+// EncryptionType specifies what encryption backend should be used (if any)
+type EncryptionType string
+
+const (
+	EncryptionTypeNone            EncryptionType = ""
+	EncryptionTypeCryptsetup      EncryptionType = "cryptsetup"
```
Comment (mvo5) [inline secboot/encrypt.go:125]: Hm, maybe this should be "LUKS" instead of "cryptsetup". What does @chrisccoulson  thing?

**Review by anonymouse64 — APPROVED** [review]
lgtm, but nitpick as always

File: secboot/encrypt.go (hunk 1 above)
Comment (mvo5) [reply secboot/encrypt.go:125]: Samuele suggested to use LUKS so I will go with that

**Pull request merged**